### PR TITLE
.Net: Rename 'Skills' -> 'Plugins' - Part 1

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Rename `Skills` to `Plugins`. First sub task from https://github.com/microsoft/semantic-kernel/issues/2119.

### Description

This is the first PR

1. Move Core, Document, MsGraph and Web skill projects to a new `Plugins` folder
2. Update Core, Document, MsGraph and Web projects to use `Plugins` instead of `Skills` in assembly names, namespace, class names
3. Extract text from `Skills.UnitTests` and create new `Plugins.UnitTests` project

Solution Explorer:
image

File Explorer:
image

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/notebooks/10-BingSearch-using-kernel.ipynb [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "#!import config/Utils.cs\n",
     "\n",
     "using Microsoft.SemanticKernel;\n",
-    "using Microsoft.SemanticKernel.Skills.Core;\n",
+    "Microsoft.SemanticKernel.Plugins.Core;\n",
```
Comment (RogerBarreto) [inline dotnet/notebooks/10-BingSearch-using-kernel.ipynb:45]: missing `using`

Comment (RogerBarreto) [inline dotnet/notebooks/10-BingSearch-using-kernel.ipynb:45]: missing `using`

```suggestion
    "using Microsoft.SemanticKernel.Plugins.Core;\n",
```

File: dotnet/notebooks/10-BingSearch-using-kernel.ipynb [hunk 2]
```diff
@@ -83,8 +83,8 @@
    },
    "outputs": [],
    "source": [
-    "using Microsoft.SemanticKernel.Skills.Web;\n",
-    "using Microsoft.SemanticKernel.Skills.Web.Bing;"
+    "Microsoft.SemanticKernel.Plugins.Web;\n",
+    "Microsoft.SemanticKernel.Plugins.Web.Bing;"
```
Comment (RogerBarreto) [inline dotnet/notebooks/10-BingSearch-using-kernel.ipynb:87]: Missing usings, are they needed ?
```suggestion
    "using Microsoft.SemanticKernel.Plugins.Web;\n",
    "using Microsoft.SemanticKernel.Plugins.Web.Bing;"
```

File: dotnet/notebooks/10-BingSearch-using-kernel.ipynb (hunk 1 above)
Comment (markwallace-microsoft) [reply dotnet/notebooks/10-BingSearch-using-kernel.ipynb]: Removing notebook changes, they shouldn't be part of this PR

Comment (markwallace-microsoft) [reply dotnet/notebooks/10-BingSearch-using-kernel.ipynb]: Removing notebook changes, they shouldn't be part of this PR

File: dotnet/notebooks/10-BingSearch-using-kernel.ipynb (hunk 2 above)
Comment (markwallace-microsoft) [reply dotnet/notebooks/10-BingSearch-using-kernel.ipynb]: Removing notebook changes, they shouldn't be part of this PR

File: dotnet/src/Skills/Skills.UnitTests/Skills.UnitTests.csproj [hunk 3]
```diff
@@ -42,12 +42,8 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\SemanticKernel\SemanticKernel.csproj" />
-    <ProjectReference Include="..\Skills.Core\Skills.Core.csproj" />
-    <ProjectReference Include="..\Skills.Document\Skills.Document.csproj" />
     <ProjectReference Include="..\Skills.Grpc\Skills.Grpc.csproj" />
```
Comment (hario90) [inline dotnet/src/Skills/Skills.UnitTests/Skills.UnitTests.csproj:45]: Should this file be renamed/removed?

**Review by hario90 — COMMENTED** [review]
Looks good - there's one csproj that looks like it needs to be removed and I can approve after the python notebooks are removed.

**Review by hario90 — APPROVED** [review]
I was reviewing old changes - looks like the notebook changes got removed.

File: dotnet/src/Skills/Skills.UnitTests/Skills.UnitTests.csproj (hunk 3 above)
Comment (markwallace-microsoft) [reply dotnet/src/Skills/Skills.UnitTests/Skills.UnitTests.csproj:45]: These two projects:

```
..\Skills.Grpc\Skills.Grpc.csproj
..\Skills.OpenAPI\Skills.OpenAPI.csproj
```

are going to be treated differently. Current thinking is that these will become:

```
..\Functions.Grpc\Functions.Grpc.csproj
..\Functions.OpenAPI\Functions.OpenAPI.csproj
```

**Pull request merged**